pkg/schedulingtypes/replicascheduler.go

```diff
@@ -273,26 +273,103 @@ func (s *ReplicaScheduler) GetSchedulingResult(rsp *fedschedulingv1a1.ReplicaSch
 		}
 	}
 
-	plnr := planner.NewPlanner(rsp)
+	plnr := planner.NewReplicaPlanner(rsp)
 	return schedule(plnr, key, clusterNames, currentReplicasPerCluster, estimatedCapacity), nil
 }
 
-func schedule(planner *planner.Planner, key string, clusterNames []string, currentReplicasPerCluster map[string]int64, estimatedCapacity map[string]int64) map[string]int64 {
+type ReplicaScheduleResult struct {
+	result map[string]int64
+}
+
+func NewReplicaScheduleResult(result map[string]int64) *ReplicaScheduleResult {
+	return &ReplicaScheduleResult{
+		result: result,
+	}
+}
+
+func (r *ReplicaScheduleResult) clusterNames() []string {
+	clusterNames := []string{}
+	for name := range r.result {
+		clusterNames = append(clusterNames, name)
+	}
+	return clusterNames
+}
+
+func (r *ReplicaScheduleResult) SetPlacementSpec(obj *unstructured.Unstructured) {
+	obj.Object[util.SpecField] = map[string]interface{}{
+		util.ClusterNamesField: r.clusterNames(),
+	}
+}
+
+// This assumes that there would be no duplicate clusternames
+func (r *ReplicaScheduleResult) PlacementUpdateNeeded(names []string) bool {
+	newNames := r.clusterNames()
+	sort.Strings(names)
+	sort.Strings(newNames)
+	return !reflect.DeepEqual(names, newNames)
+}
+
+func (r *ReplicaScheduleResult) SetOverrideSpec(obj *unstructured.Unstructured) {
+	overrides := []interface{}{}
+	for clusterName, replicas := range r.result {
+		overridesMap := map[string]interface{}{
+			util.ClusterNameField: clusterName,
+			replicasField:         replicas,
+		}
+		overrides = append(overrides, overridesMap)
+	}
+	obj.Object[util.SpecField] = map[string]interface{}{
+		util.OverridesField: overrides,
+	}
+}
+
+func (r *ReplicaScheduleResult) OverrideUpdateNeeded(typeConfig typeconfig.Interface, obj *unstructured.Unstructured) bool {
```

Comment:
The bulk of  this method is not type-specific, which suggests to me factoring out the common functionality into a type-agnostic method that can be used by both job and replica scheduling.